src/Contract/Types/VerificationKey.purs

```diff
@@ -0,0 +1,51 @@
+module HydraAuctionOffchain.Contract.Types.VerificationKey
+  ( VerificationKey
+  , vkeyBytes
+  , vkeyCodec
+  , vkeyFromBytes
+  , vkeyLength
+  ) where
+
+import Prelude
+
+import Contract.PlutusData (class FromData, class ToData)
+import Contract.Prim.ByteArray
+  ( ByteArray
+  , byteArrayFromIntArrayUnsafe
+  , byteArrayToHex
+  , byteLength
+  )
+import Data.Codec.Argonaut (JsonCodec, prismaticCodec) as CA
+import Data.Maybe (Maybe(Just, Nothing))
+import HydraAuctionOffchain.Codec (byteArrayCodec)
+import Test.QuickCheck (class Arbitrary)
+import Test.QuickCheck.Gen (chooseInt, vectorOf)
+
+newtype VerificationKey = VerificationKey ByteArray
+
+derive instance Eq VerificationKey
+derive newtype instance ToData VerificationKey
+derive newtype instance FromData VerificationKey
+
+instance Show VerificationKey where
+  show (VerificationKey vkey) = "(VerificationKey " <> byteArrayToHex vkey <> ")"
+
+instance Arbitrary VerificationKey where
+  arbitrary =
+    vectorOf vkeyLength (chooseInt 0 255)
+      <#> VerificationKey <<< byteArrayFromIntArrayUnsafe
+
+vkeyLength :: Int
+vkeyLength = 32
```

Comment:
Magic numbers should have explanatory code comments, or unbroken links to docs explaining them.